src/chatbot/bot.py

```diff
@@ -68,13 +79,170 @@ async def lmgtfy(ctx: Context) -> None:
 
 @bot.command()
 async def show_score(ctx: Context) -> None:
-    logger.info("wow")
     user_to_lookup, _ = await GamUser.objects.async_get_or_create(
         discord_id=ctx.message.author.id
     )
     await ctx.send(f"Your social score is currently {user_to_lookup.social_score}")
 
 
+@bot.command()
+async def make_prediction(
+    ctx: Context, prediction_text: str, options: Optional[str]
+) -> None:
+    if not options:
+        options = "yes,no"
+    prediction_options = options.split(",")
+    prediction_model = Prediction(prediction_text=prediction_text)
+    await prediction_model.async_save()
+    for option in prediction_options:
+        prediction_choice = PredictionChoice(prediction=prediction_model, choice=option)
+        await prediction_choice.async_save()
+    thread_message = await ctx.send(
+        f"Prediction '{prediction_text}' has been created, reply to this message with the wager command:\n!make_wager <choice> <amount>"
+    )
+    prediction_model.thread_id = thread_message.id
+    await prediction_model.async_save()
+
+
+@bot.command()
+async def make_wager(ctx: Context, choice: str, amount: int) -> None:
+    if ctx.message.reference:
+        thread_id = ctx.message.reference.message_id
+        try:
+            prediction = await Prediction.objects.async_get(thread_id=thread_id)
+            if not prediction.open:
+                logger.info(
+                    "User tried wagering on closed prediction thread_id %d", thread_id
+                )
+                await ctx.message.add_reaction(settings.WAGER_ERROR_REACTION)
+                return
+
+            user, _ = await GamUser.objects.async_get_or_create(
+                discord_id=ctx.message.author.id
+            )
+            prediction_choice = await PredictionChoice.objects.async_get(
+                choice__icontains=choice
+            )
+
+            if amount > user.gam_coins:
+                logger.info(
+                    "User %d does not have enough GamCoins to wager (tried wagering %d, only had %d)",
+                    user.discord_id,
+                    amount,
+                    user.gam_coins,
+                )
+                await ctx.message.add_reaction(settings.WAGER_ERROR_REACTION)
```

Comment:
What if we had a reaction for being out of money specifically? to disambiguate between errors and being out of cash. Maybe 💸 ? or a custom one?